### DISCUSSION TITLE
help with debugging ModuleNotFound

### DISCUSSION THREAD

**Discussion by bitterbark** [NOAA-GFDL/MDTF-diagnostics#463] [General]
I'm implementing the multirun capability into the blocking_neale pod, but failing to 'import yaml' in order to read the case_env_file. I can copy the conda activate line from my mdtf executable script and then do 'conda list' and see yaml there.  Any more ideas on how to debug?

> Found gsn_csm.ncl in /glade/work/bundy/miniconda2/envs.MDTFv3.1.20230412/_MDTF_NCL_base/lib/ncarg/nclscripts.
Traceback (most recent call last):
  File "/glade/u/home/bundy/mdtf/MDTF_3_main/MDTF-diagnostics/diagnostics/blocking_neale/blocking_neale.py", line 11, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'

(First line of above output confirms the conda env path is what is being used).

> conda activate /glade/work/bundy/miniconda2/envs.MDTFv3.1.20230412/_MDTF_base
> conda list
...
yaml                      0.2.5                h7f98852_2    conda-forge
...

**Comment by wrongkindofdoctor** [comment]
@bitterbark The yaml versions in our _MDTF_base environments are identical, but I can't replicate your issue with my own environment. Are you running the blocking_neale POD through the framework, or just testing it independently? Is pyyaml also installed? Does the same issue occur if you run a test script in the _MDTF_base environment that just imports pyyaml and prints a line or does some other simple task? 

- yaml                       0.2.5                     h7f98852_2    conda-forge (this is installed as a dependency)
- pyyaml                    6.0             py310h5764c6d_5    conda-forge (this is specified in the _MDTF_base conda env file)

**Comment by bitterbark** [comment]
Thanks for the ideas of what to check. They led me down the path to solve the problem:
I'm running an NCL-based pod, so the conda environment is _MDTF_NCL_base, not _MDTF_base. 
To fix: I added  - pyyaml=6.0 to src/conda/env_NCL_base.yml (copied from _MDTF_base file), 
ran conda_env_setup.sh --env NCL_base ...  and it works.

**Comment by wrongkindofdoctor** [comment]
@bitterbark Glad you were able to resolve the issue. I added pyyaml to the NCL_base environment file in the main branch.
